### DISCUSSION TITLE
migrating *arr (sonarr/radarr/bazarr/lidarr/prowlarr) apps from truecharts to jailmaker

### DISCUSSION THREAD

**Discussion by hnnq** [Jip-Hop/jailmaker#245] [Q&A]
I appreciate the Jellyfin example in the tutorial (and I'll have to migrate it and Jellyseerr later), but I'd like to start with migrating the *arr apps.
Is there a tutorial on how to do that
My main questions relate to:
* reusing the media datasets
* changes needed for the user/group of the media files and the *arr app (currently set to "apps:apps" = 568:568, which is the truecharts default)
* cross-referencing apps. Currently I'm using truecharts' [linking-apps](https://truecharts.org/guides/linking-apps/) to get the address of other services/apps
* I use truecharts' ingress (with traefik, blocky, as well as some other services) to access the apps by name (e.g., sonarr.mydomain), how can I accomplish the same with jailmaker
* a general question - how do I update the apps? Do I need to reinstall the docker image of the app, or is there an automatic update mechanism 

thanks

**Comment by Lockszmith-GH** [comment]
Any progress here?

Jailmaker can help you create a linux jail where you can run docker and docker-compose based containres

The way I'm planning on doing this is by creating the apps in a docker enabled jail, and write a docker-compose file for the arr stack.

I'll have another docker-compose for the reverse-proxy. Still havn't decided if it will be traefik or NPM, but I'm comfortable with either. I feel the chaining definitions in traefik will be easier for me to manage, but I do like the simplicity of the UI of NPM.

I'll setup each of these docker-compose to have their own network, with an additonal network shared between the two compose stacks.

This should make using simple host name - internally.

As for the data. I'm hoping I'll be able to copy (with send/receive) the ZFS datasets as is, and just point the docker-compose to the new location.

**Reply by hros** [reply]
Thanks. Please share your progress and insights

**Comment by ALeXXBody** [comment]
You can create a new jail and use this script to install the *arr that you need: https://github.com/pijarr/pijarr
You don't have to install docker, the *arr will work as apps under linux.
